src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/BulkTokenNumberNumberCompositeSearchParameterV1RowGenerator.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage.TvpRowGeneration
+{
+    internal class BulkTokenNumberNumberCompositeSearchParameterV1RowGenerator : BulkCompositeSearchParameterRowGenerator<(TokenSearchValue component1, NumberSearchValue component2, NumberSearchValue component3), BulkTokenNumberNumberCompositeSearchParamTableTypeV1Row>
+    {
+        private readonly BulkTokenSearchParameterV1RowGenerator _tokenRowGenerator;
+        private readonly BulkNumberSearchParameterV1RowGenerator _numberV1RowGenerator;
+
+        public BulkTokenNumberNumberCompositeSearchParameterV1RowGenerator(SqlServerFhirModel model, BulkTokenSearchParameterV1RowGenerator tokenRowGenerator, BulkNumberSearchParameterV1RowGenerator numberV1RowGenerator, SearchParameterToSearchValueTypeMap searchParameterTypeMap)
+            : base(model, searchParameterTypeMap)
+        {
+            _tokenRowGenerator = tokenRowGenerator;
+            _numberV1RowGenerator = numberV1RowGenerator;
+        }
+
+        internal override bool TryGenerateRow(
+            int offset,
+            short searchParamId,
+            (TokenSearchValue component1, NumberSearchValue component2, NumberSearchValue component3) searchValue,
+            out BulkTokenNumberNumberCompositeSearchParamTableTypeV1Row row)
+        {
+            if (_tokenRowGenerator.TryGenerateRow(offset, default, searchValue.component1, out var token1Row) &&
+                _numberV1RowGenerator.TryGenerateRow(offset, default, searchValue.component2, out var token2Row) &&
+                _numberV1RowGenerator.TryGenerateRow(offset, default, searchValue.component3, out var token3Row))
```

Comment:
@johnstairs, if you look at the diff between this file and `BulkTokenNumberNumberCompositeSearchParameterRowGenerator.cs` from the [bulk upsert branch](https://github.com/microsoft/fhir-server/compare/personal/johnstairs/prtotype-optimized-bundle-transactions), you'll see that `default` is being passed in as the first arg instead of `offset`. Is this because the token and number rows generated here are scoped to this method (ie only used in the creation of the row on line 34)?

Reply:
Yeah, we don't use the offset that is passed in to the call to _tokenRowGenerator.TryGenerateRow (we don't use token1Row.Offset), so I just made the argument default.

Reply:
Makes sense, thanks for checking. Changed in 0c52f29.